features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,588 @@
+                    <Box pt={2}>
+                      <Grid container spacing={4}>
+                        <Grid item xs={6}>
+                          <TextField
+                            fullWidth
+                            type="number"
+                            inputProps={{ min: "0" }}
+                            label={`Collateral (${collSymbol})`}
+                            error={collBalanceTooLow()}
+                            helperText={
+                              collBalanceTooLow()
+                                ? `${collSymbol} balance too low`
+                                : null
+                            }
+                            value={collateralToDeposit}
+                            onChange={(
+                              e: React.ChangeEvent<HTMLInputElement>
+                            ) => setCollateralToDeposit(e.target.value)}
+                          />
+                        </Grid>
+                        <Grid item xs={6}>
+                          <Box py={2}>
+                            {collateralToDeposit &&
+                            collateralToDeposit != "0" &&
+                            !collBalanceTooLow() ? (
+                              <Button
+                                fullWidth
+                                variant="contained"
+                                onClick={executeDeposit}
+                              >{`Deposit ${collateralToDeposit} ${collSymbol} into the position`}</Button>
+                            ) : (
+                              <Button fullWidth variant="contained" disabled>
+                                Deposit
+                              </Button>
+                            )}
+                          </Box>
+                        </Grid>
+                      </Grid>
+                      {needCollateralAllowance() && (
+                        <Typography>
+                          <br></br>
+                          <strong>Note:</strong> You will need to sign two
+                          transactions one to approve {collSymbol} and a second
+                          to preform the deposit.
+                        </Typography>
+                      )}
+                    </Box>
+                  </Box>
+                )}
+                {tabIndex === "liquidate" && (
+                  <Box pt={2}>
+                    <Typography>
+                      <strong>Liquidate this sponsor</strong>
+                      <br></br>For the position to be under collateralize{" "}
+                      {priceId ? utils.parseBytes32String(priceId) : "N/A"}{" "}
+                      would need to{" "}
+                      {underCollateralizedPrice > latestPrice
+                        ? "increase"
+                        : "decrease"}{" "}
+                      by {Math.abs(underCollateralizedPercent | 0)}% from{" "}
+                      {latestPrice?.toFixed(4)} to{" "}
```

Comment:
Do you need the ?'s after `latestPrice` and `underCollateralizedPrice` since you already check that they are non-null above?

Reply:
not anymore, you right. fixed